migrator/migrations/m_196_to_m_197_set_poduid_where_null/migration_impl.go

```diff
@@ -0,0 +1,241 @@
+	podUIDMap := make(map[string]string)
+
+	err := processIndicatorStore.Walk(ctx,
+		func(processIndicator *storage.ProcessIndicator) error {
+			_, exists := processIndicatorIds[processIndicator.Id]
+			if exists {
+				podUIDMap[processIndicator.Id] = processIndicator.GetPodUid()
+				delete(processIndicatorIds, processIndicator.Id)
+			}
+			return nil
+		})
+
+	processIndicatorIds = make(map[string]bool)
+
+	return podUIDMap, err
+}
+
+// Find out the process indicator ids of plops that could possibly be updated with a poduid from the
+// process_indicators table. This way we don't have to store all id, poduid pairs from the
+// process_indicators table. Go doesn't have sets so use a map where the value is always true instead.
+func getProcessIndicatorIdsOfInterest(ctx context.Context, plopStore plopDatastore.Store) (map[string]bool, error) {
+	processIndicatorIds := make(map[string]bool)
+
+	err := plopStore.Walk(ctx,
+		func(plop *storage.ProcessListeningOnPortStorage) error {
+			if plop.GetPodUid() == "" && plop.GetProcess() == nil {
+				processIndicatorIds[plop.GetProcessIndicatorId()] = true
+			}
+			return nil
+		})
+
+	return processIndicatorIds, err
+}
+
+// Walks through the listening_endpoints table and sets the value of PodUid where it is null using a
+// map obtained from the process_indicators table, where the key is a processindicatorid and the value
+// is a PodUid.
+func setPodUidsUsingProcessIndicators(ctx context.Context, plopStore plopDatastore.Store, podUIDMap map[string]string, batchSize int) error {
+	plops := make([]*storage.ProcessListeningOnPortStorage, batchSize)
+	count := 0
+	err := plopStore.Walk(ctx,
+		func(plop *storage.ProcessListeningOnPortStorage) error {
+			if plop.GetPodUid() == "" {
+				podUID, exists := podUIDMap[plop.GetProcessIndicatorId()]
+				if exists {
+					plop.PodUid = podUID
+					plops[count] = plop
+					count++
+				}
+
+				if count == batchSize {
+					err := plopStore.UpsertMany(ctx, plops)
+					count = 0
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		})
+
+	if count > 0 {
+		plops = plops[:count]
+		err := plopStore.UpsertMany(ctx, plops)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+// Sets PodUids by using information from the process_indicators table. It first walks through the
+// listening_endpoints table and gets processindicatorids where PodUid is null and the processindicatorid
+// could possibly be matched to an id in the process_indicators table. It then uses that to get
+// a map where the key is the processindicatorid and the value is the PodUid. It then walks through the
+// the listening_endpoints table again and sets the PodUids using the map. This is effectively a join.
+func setPodUIDsUsingProcessIndicators(ctx context.Context, processIndicatorStore processIndicatorDatastore.Store, plopStore plopDatastore.Store, batchSize int) error {
```

Comment:
Go doesn't allow function overloading. I named one function `setPodUIDsUsingProcessIndicators` and another `setPodUidsUsingProcessIndicators`. I don't know what I should call these functions.